static/js/tools/timeline_page.tsx

```diff
@@ -17,119 +17,122 @@
 import React, { Component } from "react";
 import _ from "lodash";
 import {
-  parseUrl,
   getStatsVarInfo,
   getPlaceNames,
   getStatsVar,
-  updateUrl,
+  getTimelineParamsFromUrl,
 } from "./timeline_util";
 import { SearchBar } from "./timeline_search";
 import { Menu } from "./statsvar_menu";
-import { StatsVarInfo } from "./timeline_util";
+import { StatsVarInfo, TimelineParams } from "./timeline_util";
 import { Info } from "./timeline_info";
 import { ChartRegion } from "./timeline_chart_region";
 
-interface PagePropType {
-  search: boolean;
-}
-
 interface PageStateType {
-  statsVarPaths: number[][];
+  statsVarPaths: string[][];
   statsVarInfo: { [key: string]: StatsVarInfo };
-  places: [string, string][]; // [(placeId, placeName)]
+  placeIdNames: Record<string, string>; // [(placeId, placeName)]
+  statsVarTitle: Record<string, string>;
   perCapita: boolean;
   statsVarValid: Set<string>;
 }
 
-class Page extends Component<PagePropType, PageStateType> {
-  constructor(props: PagePropType) {
+class Page extends Component<Record<string, unknown>, PageStateType> {
+  params: TimelineParams;
+
+  constructor(props: Record<string, unknown>) {
     super(props);
-    this.handleHashChange = this.handleHashChange.bind(this);
-    this._togglePerCapita = this._togglePerCapita.bind(this);
-    this.setStatsVarTitle = this.setStatsVarTitle.bind(this);
+    this.params = getTimelineParamsFromUrl();
+    const tempStatsVarTitle = {};
```

Comment:
can just call this statsVarTitle; temp is for temporary variable while this one is actually used (and later it is override)

Reply:
got it.